doc/clientTypeMigrations.md

```diff
@@ -0,0 +1,33 @@
+# Migrating managed client types to Grpc.Core.Api
+
+Migrating types for managed client will be different from types required for managed server. The main difference is that there’s a default client base implementation whose functionality must be maintained for back compat. The types `ClientBase<T>` and `ClientBase` has constructors that take in Channel which contains abstractions for making native calls using c core which is then used by the `DefaultCallInvoker`. It is difficult to decouple `ClientBase` and `Channel` while maintaining current functionality. As an alternative, it may be necessary to modify codegen to use a different base class for the client.
+
+To elaborate:
+1. Create a new client base class `ManangedClientBase<T>` which doesn’t have the constructor overload that takes in a `Channel` but is otherwise similar to `ClientBase<T>`.
+2. The constructor overload that takes in a `CallInvoker` will be used in the managed client implementation
+3. Additional types potentially needed include:
+
+* UnimplementedCallInvoker
+  * Needed for parameterless  constructor of `ManagedClientBase<T>`
+* AsyncClientStreamingCall<TResponse>
+* AsyncDuplexStreamingCall<TResponse>
+* AsyncServerStreamingCall<TResponse>
+* AsyncUnaryCall<TResponse>
+* CallInvoker
+* IClientStreamWriter<T>
+* CallFlags
+  * Internal class which may not be needed
+* CallCredentials?
+  * This type may not be needed as it seems like it’s used to add `Authorization` headers with a per call granularity. We provide functionality through the APIs on `HttpClient` to do this but for portability reasons, though tenuous, we could move this type too. If we do decide to move this type, we need to change the implementation and remove the ToNativeCredentials() method.
```

Comment:
the problem is that CallOptions depend on CallCredentials (and it feels odd if the "credentials" as one of the CallOptions' constructor param would be unimplemented).
How hard would be to implement CallCredentials over HttpClient?
Another problem is that CallCredentials has some public static factory methods that transitively pull in lot of the native functionality.

Reply:
I think I have a reasonable solution for migrating CallCredentials to Grpc.Core.Api, I think I will have a PR soon.